embedding_handler.py

```diff
@@ -0,0 +1,137 @@
+        model.compile(loss='binary_crossentropy', optimizer='rmsprop', metrics=['accuracy'])
+        self.compiled_network = model
+        #Here we define a secondary model which we can pass a pair of heroes into to get a measure 
+        #of how similar their embeddings are (bigger number = more similar).
+        cos_similarity_layer = keras.layers.dot(
+            inputs=[target_hero_embedding, teammate_hero_embedding],
+            axes=0,
+            normalize=True
+        )
+        self.cos_sim_model = keras.Model([target_hero_input, teammate_hero_input], cos_similarity_layer)
+        
+        print(model.summary())
+        print("Model succesfully assembled, pass self.load_data(data) to prepare your data for training")
+
+    def load_odota_data(self, data):
+        data = list(data)
+        #The summary data we pulled from the open dota api gave the heroes as a single string with the 
+        #api keys separated by commas, so we need to convert this string of api keys to a list of integers
+        #that correspond to the index of the hero
+        data = [team.split(",") for team in data]
+        for team in range(len(data)):
+            data[team] = [int(hero) for hero in data[team]]
+        for i, team in enumerate(data):
+            for j, hero in enumerate(team):
+                data[i][j] = self.api_to_nn[hero]
+        couples_list = []
+        labels_list = []
+        #the skipgrams keras function generates a sequence of pairs from the supplied data and labels it 1 or 0
+        #where a pair labeled 1 the teammate was on the target hero's team, and 0 if the pair did not appear in
+        #in the target hero's team
+        for team in data:
+            couples, labels = keras.preprocessing.sequence.skipgrams(
+                sequence=team,
+                vocabulary_size=len(self.api_to_nn),
+                window_size=len(team),
+                negative_samples=1
+            )
+            couples_list += couples
+            labels_list += labels
+        target_hero, teammate_hero = zip(*couples_list)
+        self.target_hero = np.array(target_hero, dtype="int32")
+        self.teammate_hero = np.array(teammate_hero, dtype="int32")
+        self.labels = labels_list
+        print("Data successfully loaded into model object, pass self.train_model() to train the model")
+    
+    def train_model(self, num_in_batch, num_epochs):
+        self.history = self.compiled_network.fit((self.target_hero, self.teammate_hero), self.labels,
+                                 batch_size=num_in_batch,
+                                 epochs=num_epochs,
+                                 validation_split=0.1)
+        print("Training complete")
+        
+    def k_closest_heroes(self, hero_name, k):
+        #Prints the k most similar heroes to the input hero_name according to their cosine similarity,
+        #NOT according to euclidean distance
```

Comment:
When you have a method which you would like to explain in a comment what it does, you should use a `docstring` instead of a comment, so editors pick it up for autocomplete.

A docstring is a string literal placed directly below the start of a module, class, or method like this:

```
class Animal():
    """ An abstract animal. """
    
    def makeNoise(self):
        """ Prints out the noise this animal makes, depending on what kind of animal it is. """
        pass
```

VS Code and other editors will show this docstring when you have an `Animal` and you select `makeNoise` from the autocomplete suggestion list, acting as documentation.